ui/src/features/WorkspaceSettings/components/GeneralSettings.tsx

```diff
@@ -6,25 +6,44 @@ import { useWorkspace } from "@flow/lib/gql";
 import { useT } from "@flow/lib/i18n";
 import { useCurrentWorkspace } from "@flow/stores";
 
+type Errors = "delete" | "update";
+
 const GeneralSettings: React.FC = () => {
   const t = useT();
-  const [currentWorkspace] = useCurrentWorkspace();
-  const { deleteWorkspace } = useWorkspace();
+  const [currentWorkspace, setCurrentWorkspace] = useCurrentWorkspace();
+  const { deleteWorkspace, updateWorkspace } = useWorkspace();
   const navigate = useNavigate();
-  const [showError, setShowError] = useState(false);
+  const [showError, setShowError] = useState<Errors | undefined>(undefined);
+  const [workspaceName, setWorkspaceName] = useState(currentWorkspace?.name);
+  const [loading, setLoading] = useState(false);
 
   const handleDeleteWorkspace = async () => {
-    setShowError(false);
+    setLoading(true);
```

Comment:
Sholdn't be get a `loading` or `isLoading` variable from the gql api that we can use instead of locally setting state? 🤔

Reply:
That only works for `queries`. for mutations we are using `mutateAsync` which returns a promise. We can use the `isLoading` if we use the `mutate` function but then it'll go inside a hook and not a normal function.

Reply:
Ok. Just keep in mind places where you are doing this, since when we add optimistic updates this probably will become unnecessary 🙏🏻 
When you read this you can resolve this convo yourself 👍🏻